pkg/shared/utils/files/files.go

```diff
@@ -0,0 +1,52 @@
+package files
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// Walk walks the file tree rooted at root including symlinks, calling fn for
+// each file in the tree.
+func Walk(root string, fn filepath.WalkFunc) error {
+	return walk(root, "", fn)
+}
+
+func walk(filename string, linkDirName string, fn filepath.WalkFunc) error {
+	if linkDirName == "" {
+		linkDirName = filename
+	}
+
+	symWalkFunc := func(path string, info os.FileInfo, err error) error {
+		if fname, err := filepath.Rel(filename, path); err == nil {
+			path = filepath.Join(linkDirName, fname)
+		} else {
+			return err
+		}
+
+		if err == nil && info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			finalPath, err := filepath.EvalSymlinks(path)
+			if err != nil {
+				return err
+			}
+
+			info, err := os.Lstat(finalPath)
+			if err == nil {
```

Comment:
Hey, I'm aware this has been merged, but I'm just wondering if these lines were intentional as in:
```golang
if err == nil {
  return nil
}
```
or a typo for:
```diff 
- if err = nil {
+ if err != nil {
```

Reply:
smells like a typo

Reply:
Ok, I should warn there are some tests failing when the change is made